client/src/main/java/org/eclipse/hono/client/impl/AmqpAdapterClientCommandResponseSender.java

```diff
@@ -0,0 +1,131 @@
+     *
+     * @param con The connection to the Hono server.
+     * @param tenantId The tenant that the events will be published for.
+     * @param remoteCloseHook The handler to invoke when the link is closed by the peer (may be {@code null}). The
+     *            sender's target address is provided as an argument to the handler.
+     * @return A future indicating the outcome.
+     * @throws NullPointerException if con or tenantId is {@code null}.
+     */
+    public static Future<CommandResponder> createWithAnonymousLinkAddress(
+            final HonoConnection con,
+            final String tenantId,
+            final Handler<String> remoteCloseHook) {
+
+        Objects.requireNonNull(con);
+        Objects.requireNonNull(tenantId);
+
+        return con.createSender(null, ProtonQoS.AT_LEAST_ONCE, remoteCloseHook)
+                .map(sender -> new AmqpAdapterClientCommandResponseSender(con, sender, tenantId));
+    }
+
+    @Override
+    public Future<ProtonDelivery> sendCommandResponse(final String deviceId, final String targetAddress,
+            final String correlationId, final int status, final byte[] payload, final String contentType,
+            final Map<String, ?> properties) {
+        return sendCommandResponse(deviceId, targetAddress, correlationId, status, payload, contentType,
+                properties, null);
+    }
+
+    @Override
+    public Future<ProtonDelivery> sendCommandResponse(final String deviceId, final String targetAddress,
+            final String correlationId, final int status, final byte[] payload,
+            final String contentType, final Map<String, ?> properties, final SpanContext context) {
+
+        Objects.requireNonNull(deviceId);
+        Objects.requireNonNull(targetAddress);
+        Objects.requireNonNull(correlationId);
+
+        final Message message = ProtonHelper.message();
+        message.setAddress(targetAddress);
+        message.setCorrelationId(correlationId);
+
+        MessageHelper.setCreationTime(message);
+        MessageHelper.addProperty(message, MessageHelper.APP_PROPERTY_TENANT_ID, tenantId);
+        MessageHelper.addProperty(message, MessageHelper.APP_PROPERTY_DEVICE_ID, deviceId);
+        MessageHelper.addProperty(message, MessageHelper.APP_PROPERTY_STATUS, status);
+        MessageHelper.setPayload(message, contentType, payload);
+
+        setApplicationProperties(message, properties);
+
+        return sendAndWaitForOutcome(message, context);
+    }
+
+    /**
+     * This method must not be used outside a protocol adapter.
+     * 
+     * @param commandResponse not used.
+     * @param context not used.
+     * @return nothing.
+     * @throws UnsupportedOperationException always.
+     */
+    @Override
+    public Future<ProtonDelivery> sendCommandResponse(final CommandResponse commandResponse,
+            final SpanContext context) {
+        throw new UnsupportedOperationException("This method must not be used outside a protocol adapter");
```

Comment:
I think the exception text and the method javadoc is confusing here - if the class and method is used inside a protocol adapter, there is still this exception.
I guess the javadoc could state that the _overridden method_ is only meant for use by a protocol adapter to send the command response downstream, and that this use case isn't supported for this class.